File: src/main/java/ru/dargen/evoplus/feature/setting/StringSetting.java
package ru.dargen.evoplus.feature.setting;

import com.google.gson.JsonElement;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import ru.dargen.evoplus.EvoPlus;
import ru.dargen.evoplus.gui.GuiElement;

import java.util.function.Consumer;

@Builder
@Getter
@AllArgsConstructor
public class StringSetting implements Setting<String> {

    @Builder.Default
    protected String value = "";
    protected final String name;
    protected final String id;
    @Builder.Default
    protected final Consumer<StringSetting> changeHandler = null;

    @Override
    public GuiElement getElement() {
        return null;
    }

    public void setValue(String value) {
        this.value = value;
        if (changeHandler != null) changeHandler.accept(this);
    }

    @Override
    public void load(JsonElement jsonElement) {
        setValue(jsonElement.getAsString());
    }

    @Override
    public JsonElement unload() {
        return EvoPlus.instance().getGson().toJsonTree(value);
    }

    @Override
    public boolean isDrawable() {
        return false;
    }

}

File: src/main/kotlin/ru/dargen/evoplus/feature/settings/StringSetting.kt
package ru.dargen.evoplus.feature.settings

import com.google.gson.JsonElement
import com.google.gson.JsonPrimitive
import ru.dargen.evoplus.api.render.node.Node
import ru.dargen.evoplus.api.render.node.input.button
import ru.dargen.evoplus.util.asBoolean

class StringSetting(id: String, name: String, value: String) : Setting<String>(id,name) {

    override var value: String = value
        set(value) {
            field = value
            handler(value)
        }

    override val settingElement
        get() = button(value)

    override fun load(element: JsonElement) {
        value = element.asString
    }

    override fun store() = JsonPrimitive(value)
}